### ISSUE TITLE
Deprecate the `dag_longest_path_length` helper function?

### ISSUE THREAD

**Issue by rossbar** [networkx/networkx#6383]
The `nx.dag_longest_path` function has a helper `dag_longest_path_length` which essentially calls `dag_longest_path` then does a little post-processing to derive the length from the result. This generally matches a pattern established by the `shortest_path` functions (i.e. having a `_length` helper), but I thought I'd make the case for breaking that pattern here.

For non-multigraphs, `nx.dag_longest_path_length` essentially boils down to:

```python
sum(G[u][v].get(weight, default_weight) for u, v in pairwise(path))
```

where `path` is the result of `nx.dag_longest_path`. This one-liner is idiomatic and easy to understand (and would be simplified even further in the case of e.g. unweighted edges). The case of multigraphs is trickier, as there is the matter of how to reduce weighted multiedges into a single value. `dag_longest_path_length` makes the (sensible) decision to use `max` to select the maximum edge weight. There's no reason to be this rigid however - users should be able to use whatever reduction function they want (`sum` e.g.) and IMO it's easier to bake in this flexibility by using Python comprehensions directly rather than expanding the API to support passing in a `callable`.

Like most of my deprecation suggestions, this is very subjective. I prefer a teach-a-person-to-fish approach when dedicated API can generally be replaced with a one-line list comprehension, but that has to be weighed against the potential disruption from deprecating, as well as the breaking of the `_length` helper-function API pattern established by the `shortest_path` package. Please LMK what you think!

See also #6311

**Comment by dschult** [comment]
I agree that if the `*path_length` version of the function simply calculates the path and then from that computes the length that there isn't a good argument for a NetworkX function providing the length.  And example in the `*path` function should be sufficient.  That is the case with the `dag_longest_path_length` function.  The argument for the `*_path_length` functions in the `shortest_path` suite of functions is that it saves a bunch of time and memory if you don't have to collect the paths...  If the length is all you are after, then you can compute the length more efficiently. 

In summary, it seems like the `*_length` functions should be provided when the algorithm to compute them is significantly more efficient than computing the paths followed by the length of those paths.

Thanks for bringing this up!

**Comment by Qudirah** [comment]
@MridulS this is solved. The issue should be closed.

**Comment by dschult** [comment]
This will be automatically closed when the associated PR is merged. (the link between the two is shown in the right sidebar on github under the heading "Development".
